photon-ml/src/main/scala/com/linkedin/photon/ml/function/svm/IndividualSmoothedHingeLossFunction.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.function.svm
+
+import breeze.linalg.Vector
+import org.apache.spark.broadcast.Broadcast
+
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.function.{DiffFunction, IndividualObjectiveFunction, L2RegularizationDiff}
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.optimization.{GLMOptimizationConfiguration, RegularizationType}
+import com.linkedin.photon.ml.util.Utils
+
+/**
+ * This class is used to calculate the value and gradient of Rennie's smoothed hinge loss function, as an approximation
+ * of a linear SVM, for individual optimization problems.
+ *
+ * FAQ: Why use cumGradient (cumulative gradient)?
+ * A:   Using cumGradient allows the functions to avoid memory allocation by modifying and returning cumGradient instead
+ *      of creating a new gradient vector.
+ */
+protected[ml] class IndividualSmoothedHingeLossFunction extends IndividualObjectiveFunction with DiffFunction {
```

Comment:
This "Individual" - I don't find very clear. I would prefer "SingleHost" or "Local". "Local" might be the best, people are familiar with "local/distributed". "Individual" to me is connected to "one" like "individual/group", so it's not terribly appropriate here, I think.